sentinelhub/api/batch/statistical.py

```diff
@@ -0,0 +1,215 @@
+"""
+Module implementing an interface with
+`Sentinel Hub Batch Processing API <https://docs.sentinel-hub.com/api/latest/api/batch-statistical/>`__.
+"""
+import datetime as dt
+import logging
+import sys
+from dataclasses import dataclass, field
+from typing import Any, List, Optional, Union
+
+from dataclasses_json import CatchAll, LetterCase, Undefined
+from dataclasses_json import config as dataclass_config
+from dataclasses_json import dataclass_json
+
+from ...type_utils import Json, JsonDict
+from ..base_request import InputDataDict
+from ..statistical import SentinelHubStatistical
+from ..utils import datetime_config, enum_config, remove_undefined
+from .base import BaseBatchClient, BaseBatchRequest, BatchRequestStatus, BatchUserAction
+
+if sys.version_info < (3, 11):
+    from typing_extensions import NotRequired, TypedDict
+else:
+    from typing import NotRequired, TypedDict  # pylint: disable=ungrouped-imports
+
+LOGGER = logging.getLogger(__name__)
+
+BatchStatisticalRequestType = Union[str, dict, "BatchStatisticalRequest"]
+
+
+class S3AccessSpecification(TypedDict):
+    """Specification of a S3 path."""
+
+    url: str
+    accessKey: str
+    secretAccessKey: str
+    region: NotRequired[str]
+
+
+class S3Specification(TypedDict):
```

Comment:
Shouldn't this class be called maybe `StorageSpecification` or `AccessSpecification` and the one above `S3Specification`? This is because here `s3` is only one of the parameters (atm the only one).

Reply:
done